src/main/java/strategy/PurchaseStrategy.java

```diff
@@ -0,0 +1,10 @@
+package strategy;
+
+import model.FruitTransaction;
+
+public class PurchaseStrategy implements TransactionStrategy {
+    @Override
+    public void apply(FruitTransaction fruit, FruitTransaction transaction) {
+        fruit.setQuantity(Math.max(0, fruit.getQuantity() - transaction.getQuantity()));
```

Comment:
```suggestion
        fruit.setQuantity(fruit.getQuantity() - transaction.getQuantity());
```
Why do you need this method? What happens if fruits aren’t enough?

Reply:
If the quantity of Purchase strategy is bigger than stored fruits quantity - setting a negative quantity will slow down the store's calculations. 

This prevents sales capacity from being overwhelmed.